src/common/openfam_wrappers.cpp

```diff
@@ -0,0 +1,1836 @@
+    }
+}
+
+c_fam_region_desc* c_fam_create_region(c_fam* fam_obj, const char* fam_region,
+                                       size_t size, mode_t perm, c_fam_Region_Attributes* reg_att) {
+    Region* region_desc = nullptr;
+    fam* fam_inst = (fam*) fam_obj;
+    try {
+        region_desc = fam_inst->fam_create_region(fam_region, size, perm,
+                                                  (Fam_Region_Attributes*)reg_att);
+    } catch (openfam::Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return (c_fam_region_desc*)region_desc;
+    }
+    return (c_fam_region_desc*)region_desc;
+}
+
+c_fam_region_desc* c_fam_create_region_ctx(c_fam* fam_obj, const char* fam_region,
+                                       size_t size, mode_t perm, c_fam_Region_Attributes* reg_att) {
+    Region* region_desc = nullptr;
+    fam_context* fam_inst = (fam_context*) fam_obj;
+    try {
+        region_desc = fam_inst->fam_create_region(fam_region, size, perm,
+                                                  (Fam_Region_Attributes*)reg_att);
+    } catch (openfam::Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return (c_fam_region_desc*)region_desc;
+    }
+    return (c_fam_region_desc*)region_desc;
+}
+
+c_fam_region_desc* c_fam_lookup_region(c_fam* fam_obj, const char* name) {
+    Region* region_desc = NULL;
+    fam* fam_inst = (fam*) fam_obj;
+    try {
+        region_desc = fam_inst->fam_lookup_region(name);  
+    } catch (openfam::Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return (c_fam_region_desc*)region_desc;
+    }
+    return (c_fam_region_desc*)region_desc;
+}
+
+c_fam_region_desc* c_fam_lookup_region_ctx(c_fam* fam_obj, const char* name) {
+    Region* region_desc = NULL;
+    fam_context* fam_inst = (fam_context*) fam_obj;
+    try {
+        region_desc = fam_inst->fam_lookup_region(name);  
+    } catch (openfam::Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return (c_fam_region_desc*)region_desc;
+    }
+    return (c_fam_region_desc*)region_desc;
+}
+
+int c_fam_destroy_region(c_fam* fam_obj, c_fam_region_desc* region) {
+    fam* fam_inst = (fam*) fam_obj;
+    try {
+        fam_inst->fam_destroy_region((Region*)region);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return -1;
+    }
+    return 0;
+}
+
+int c_fam_destroy_region_ctx(c_fam* fam_obj, c_fam_region_desc* region) {
+    fam_context* fam_inst = (fam_context*) fam_obj;
+    try {
+        fam_inst->fam_destroy_region((Region*)region);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return -1;
+    }
+    return 0;
+}
+
+int c_fam_finalize(c_fam* fam_obj, const char* groupname) {
+    fam* fam_inst = (fam*) fam_obj;
+    try {
+        fam_inst->fam_finalize(groupname);
+    } catch (Fam_Exception &e) {
+        CAPTURE_EXCEPTION(e);
+        return -1;
+    }
+    return 0;
+}
+
+int c_fam_finalize_ctx(c_fam* fam_obj, const char* groupname) {
```

Comment:
What is fam_finalize_ctx ? why is there a printf at the end of this function

Reply:
Removed the trace prints. Added comment for the function.